### DISCUSSION TITLE
FindAsync for multiple requests

### DISCUSSION THREAD

**Discussion by jakSolutionsGithub** [fuzzzerd/fmdata#203] [Q&A] [answered]
Hey,

I'm trying to perform a find with 2 differents requests, override the default limit and include a sort order but I didn't understand how should I do that with fmdata and the FindRequest() method.

Let's say I'm trying to build a find request where 'Group', 'Work State' and 'FirstName' are properties of the model Employee in my .NET project  :

```
  [DataContract(Name = "API_employees")]
  public class Employee
  {
      [DataMember(Name = "Group")]
      public string Group { get; set; }

      [DataMember(Name = "WorkState")]
      public string WorkState { get; set; }

      [DataMember(Name = "FirstName")]
      public string FirstName { get; set; }
  }
```

How should I use .FindAsync() to build this query?

```
{
"limit":500,
 "query":[
  {"Group": "=Surgeon", "Work State" : "CA"},
  {"Group": "=Surgeon", "Work State" : "NY", "omit" : "true"}],
 "sort":[
  {"fieldName": "Work State","sortOrder": "ascend"},
  {"fieldName": "FirstName", "sortOrder": "ascend"} ]
}
```

Thanks a lot in advance!

**Comment by fuzzzerd** [comment, marked as answer]
That is a more advanced use case, and the `.FindAsync()` method is not designed for what I'd call an "advanced" find operation; however, the Data API and FMData should be able to generate the API calls that you need.

To perform multiple find requests, you will use the `.SendAsync()` overload that takes an `IFindRequest`. If you poke around the source, you'll notice that the `.FindAsync()` method actually just creates an `IFindRequest` and passes it along to `.SendAsync()` so that's worth checking out as one example: https://github.com/fuzzzerd/fmdata/blob/9687e7033a47970851fa75b179207f551d5ddeff/src/FMData/FileMakerApiClientBase.cs#L378-L394.

An example targeted for your post would be something like below:

I want to emphasize this point though, the way the sorting is handled is going to change in v5:https://github.com/fuzzzerd/fmdata/issues/158. It will be a breaking change, so watch for that. No planned release of v5, but its something I'm chipping away at.

```cs
var findOne = new Employee { Group = "=Surgeon", WorkState = "CA" };
var findTwo = new Employee { Group = "=Surgeon", WorkState = "NY" };
var request = _fmdata.GenerateFindRequest<Employee>.UseLayout(new Employee());

// note the call to AddQuery on the FindRequest object.
// this lets you specify which find requests are omit style and which are not
request.AddQuery(query: findOne, omit: false);
request.AddQuery(query: findTwo, omit: true);

// sort
request.Sort = new[]
{
    new FMData.Rest.Requests.Sort()
    {
        SortField = "Work State", // actual filemaker field name used here
        SortOrder = "ascend"
    },
    new FMData.Rest.Requests.Sort()
    {
        SortField = "FirstName", // actual filemaker field name used here
        SortOrder = "ascend"
    },
};

// execute the find and load the results
var results = _fmdata.SendAsync(request);

```
